patches/net/minecraft/world/entity/LivingEntity.java.patch

```diff
@@ -330,23 +369,48 @@
                      }
                  }
              }
-@@ -1699,6 +_,8 @@
+@@ -1690,6 +_,7 @@
+                 int k = EnchantmentHelper.getDamageProtection(this.getArmorAndBodyArmorSlots(), p_21193_);
+                 if (k > 0) {
+                     p_21194_ = CombatRules.getDamageAfterMagicAbsorb(p_21194_, (float)k);
++                    this.damageContainer.setEnchantReduction(this.damageContainer.getNewDamage() - p_21194_);
+                 }
+ 
+                 return p_21194_;
+@@ -1698,23 +_,26 @@
+     }
  
      protected void actuallyHurt(DamageSource p_21240_, float p_21241_) {
-         if (!this.isInvulnerableTo(p_21240_)) {
-+            p_21241_ = net.neoforged.neoforge.common.CommonHooks.onLivingHurt(this, p_21240_, p_21241_);
-+            if (p_21241_ <= 0) return;
-             p_21241_ = this.getDamageAfterArmorAbsorb(p_21240_, p_21241_);
-             p_21241_ = this.getDamageAfterMagicAbsorb(p_21240_, p_21241_);
-             float f1 = Math.max(p_21241_ - this.getAbsorptionAmount(), 0.0F);
-@@ -1708,6 +_,7 @@
+-        if (!this.isInvulnerableTo(p_21240_)) {
```

Comment:
Why was the invulnerable check removed here?  It is possible to invoke `actuallyHurt` without going through `hurt`, so the check should remain here.

Reply:
`actuallyHurt` is a protected method and is only ever invoked within `hurt` which is where we first check the invulnerability.  The removal was to remain consistent throughout the pipeline and ensure invulnerability check happens at the very beginning, isn't repeatedly invoked, and properly treats `actuallyHurt` as a final application of damage for the purpose of clarity.

Reply:
after discord discussion this call has been restored